File: front/lib/zhilian-filters.ts
export type FilterNode = { code: string; name: string; children: FilterNode[] }
export type ZhilianFilters = {
  district?: string; subwayLine?: string; subwayStation?: string;
  education?: string[]; experience?: string[]; companyType?: string[]; financing?: string[];
  companySize?: string[]; workNature?: string[]; jobCategory?: string; industry?: string[];
}
export type FilterCatalog = { source: string; version: string; cityCode: string; cityName: string; options: Record<string, FilterNode[]> }
export const MULTI_FILTERS = [
  ['education','学历'],['experience','经验'],['companyType','公司性质'],
  ['financing','融资阶段'],['companySize','公司人数'],['workNature','工作性质'],
] as const
export function resetCityFilters(filters: ZhilianFilters = {}): ZhilianFilters {
  return {...filters,district:'',subwayLine:'',subwayStation:''}
}
export function findFilterPath(nodes: FilterNode[], code: string): FilterNode[] {
  for (const node of nodes) {
    if (node.code === code) return [node]
    const path = findFilterPath(node.children || [], code)
    if (path.length) return [node,...path]
  }
  return []
}

File: src/main/java/com/getjobs/application/dto/ZhilianFilters.java
package com.getjobs.application.dto;

import com.fasterxml.jackson.annotation.JsonAnySetter;
import lombok.Data;
import java.util.ArrayList;
import java.util.List;

/** Official Zhilian codes, not BOSS codes. Empty values mean no restriction. */
@Data
public class ZhilianFilters {
    private String district = "";
    private String subwayLine = "";
    private String subwayStation = "";
    private List<String> education = new ArrayList<>();
    private List<String> experience = new ArrayList<>();
    private List<String> companyType = new ArrayList<>();
    private List<String> financing = new ArrayList<>();
    private List<String> companySize = new ArrayList<>();
    private List<String> workNature = new ArrayList<>();
    private String jobCategory = "";
    private List<String> industry = new ArrayList<>();

    @com.fasterxml.jackson.annotation.JsonIgnore
    public boolean isActive() {
        return java.util.stream.Stream.of(district,subwayLine,subwayStation,jobCategory).anyMatch(v->v!=null&&!v.isBlank())
                || java.util.stream.Stream.of(education,experience,companyType,financing,companySize,workNature,industry).anyMatch(v->v!=null&&!v.isEmpty());
    }

    @JsonAnySetter
    public void rejectUnknown(String name, Object value) {
        throw new IllegalArgumentException("不支持的智联筛选字段：" + name);
    }
}
